core/src/main/java/com/gentics/mesh/plugin/manager/MeshPluginManagerImpl.java

```diff
@@ -588,4 +574,55 @@ public PluginResponse toResponse(MeshPlugin plugin) {
 		return response;
 	}
 
+	protected void startPluginsFrom(List<PluginWrapper> source) {
+		List<PluginWrapper> sourceCopy;
+		
+		if (source == failedPlugins) {
+			sourceCopy = new ArrayList<>(source);
+		} else {
+			sourceCopy = source;
+		}
+		
+		for (PluginWrapper pluginWrapper : sourceCopy) {
+			PluginState pluginState = pluginWrapper.getPluginState();
+			if ((PluginState.DISABLED != pluginState) && (PluginState.STARTED != pluginState)) {
+				try {
+					log.info("Start plugin '{}'", getPluginLabel(pluginWrapper.getDescriptor()));
+
+					Plugin plugin = pluginWrapper.getPlugin();
+
+					//if (plugin.isUsingRestClient()) {
+						database.blockingTopologyLockCheck();
+					//}
+
+					plugin.start();
+					// Set state for PF4J
+					pluginWrapper.setPluginState(PluginState.STARTED);
+					if (plugin instanceof MeshPlugin) {
+						// Set status for Mesh
+						MeshPlugin meshPlugin = (MeshPlugin) plugin;
+						setStatus(meshPlugin.id(), PluginStatus.STARTED);
+						pluginRegistry.preRegister(meshPlugin);
+					}
+					startedPlugins.add(pluginWrapper);
+					
+					if (source == failedPlugins) {
+						source.remove(pluginWrapper);
+					}
+					
+					firePluginStateEvent(new PluginStateEvent(this, pluginWrapper, pluginState));
+				} catch (Throwable e) {
+					if (source == failedPlugins) {
+						log.error("Error while starting plugin {" + pluginWrapper.getPluginId() + "}: " + e.getMessage(), e);
+					} else {
+						log.warn("First error while starting plugin {" + pluginWrapper.getPluginId() + "}: " + e.getMessage() + ", giving a retry...", e);
+						failedPlugins.add(pluginWrapper);
```

Comment:
This approach to reinit failed plugins is IMO too basic, so if the reinit really helps and does not harm the overall Mesh startup process, I would make a more sophisticated version of it, including configs, timeouts and killswitch.